### PR TITLE
fix(item-action): add aria-label to plainbutton

### PR DESCRIPTION
**Pull request by jpan-box**
**After**
* aria-label added, matching hover text
* aria-describedby removed
Screenshot 2022-12-13 at 4 56 16 PM
Screenshot 2022-12-14 at 10 56 01 AM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
         "lint:js": "eslint --max-warnings=0 .",
         "lint:ts": "tsc && eslint --ext=.tsx,.ts --max-warnings=0 .",
         "lint:css": "stylelint \"src/**/*.scss\" --syntax scss",
-        "prebuild:es": "npm-run-all build:variables copy:styles ts:defs copy:flow",
```
Comment (tjuanitas) [inline package.json:69]: let's not commit this

File: src/elements/content-uploader/ItemAction.js [hunk 2]
```diff
@@ -101,7 +101,12 @@ const ItemAction = ({
         <div className="bcu-item-action">
             {tooltip ? (
                 <Tooltip position="top-left" text={intl.formatMessage(tooltip)}>
-                    <PlainButton onClick={onClick} type="button" isDisabled={status === STATUS_STAGED}>
+                    <PlainButton
+                        onClick={onClick}
+                        type="button"
+                        isDisabled={status === STATUS_STAGED}
+                        aria-label={intl.formatMessage(tooltip)}
```
Comment (tjuanitas) [inline src/elements/content-uploader/ItemAction.js:108]: nit: could we alphasort these props?

File: src/elements/content-uploader/__tests__/ItemAction.test.js [hunk 3]
```diff
@@ -15,7 +16,7 @@ describe('elements/content-uploader/ItemAction', () => {
     const getWrapper = props =>
         shallow(
             <ItemAction
-                intl={{ formatMessage: data => <span {...data} /> }}
+                intl={{ formatMessage: data => data.defaultMessage }}
```
Comment (tjuanitas) [inline src/elements/content-uploader/__tests__/ItemAction.test.js:19]: good catch!

**Review by tjuanitas — COMMENTED** [review]
looks great overall! small things

could we include similar changes for `ItemRemove` as well?

File: src/elements/content-uploader/__tests__/ItemAction.test.js (hunk 3 above)
Comment (jpan-box) [reply src/elements/content-uploader/__tests__/ItemAction.test.js:19]: it was didi :D

File: package.json (hunk 1 above)
Comment (jpan-box) [reply package.json:69]: ok!

File: src/elements/content-uploader/ItemAction.js (hunk 2 above)
Comment (jpan-box) [reply src/elements/content-uploader/ItemAction.js:108]: yes, i need to think about things that way!

Comment (tjuanitas) [reply src/elements/content-uploader/ItemAction.js:108]: don't worry bout it, we'll add a prettier config to handle this

**Comment by jpan-box** [comment]
Planning on handling ItemRemove in a separate PR!

File: src/elements/content-uploader/ItemAction.js [hunk 4]
```diff
@@ -101,7 +101,12 @@ const ItemAction = ({
         <div className="bcu-item-action">
             {tooltip ? (
                 <Tooltip position="top-left" text={intl.formatMessage(tooltip)}>
-                    <PlainButton onClick={onClick} type="button" isDisabled={status === STATUS_STAGED}>
+                    <PlainButton
+                        aria-label={intl.formatMessage(tooltip)}
```
Comment (tjuanitas) [inline src/elements/content-uploader/ItemAction.js:105]: not necessary to change but could destruct `intl` near the top of the function:

`const { formatMessage } = intl;`

**Review by tjuanitas — APPROVED** [review]
LGTM

File: src/elements/content-uploader/ItemAction.js [hunk 5]
```diff
@@ -101,7 +101,12 @@ const ItemAction = ({
...
+                        isDisabled={status === STATUS_STAGED}
```
Comment (karelee7) [inline src/elements/content-uploader/ItemAction.js:106]: super minor and not absolutely necessary! But for readability, should we put this in a variable rather than inline it? (haven't seen this inlining pattern for props that often elsewhere)

Comment (jpan-box) [reply src/elements/content-uploader/ItemAction.js:106]: I'm all for making things more readable! I only see my perspective so it's helpful to hear where things can be changed :D Just to make sure, do you mean aria-label should be defined as a variable or this `status === STATUS_STAGED`?

File: src/elements/content-uploader/ItemAction.js (hunk 4 above)
Comment (jpan-box) [reply src/elements/content-uploader/ItemAction.js]: did it!

File: src/elements/content-uploader/ItemAction.js (hunk 5 above)
Comment (jpan-box) [reply src/elements/content-uploader/ItemAction.js]: I think I did both with @tjuanitas 's help!

**Review by tjuanitas — APPROVED** [review]
lgtm

**Review by QingyuChai — APPROVED** [review]
l g t m

File: src/elements/content-uploader/ItemAction.js [hunk 6]
```diff
@@ -50,6 +50,7 @@ const ItemAction = ({
     let icon = <IconClose />;
     let tooltip;
     const { code } = error;
+    const { formatMessage } = intl;
```
Comment (JChan106) [inline src/elements/content-uploader/ItemAction.js:53]: can we still use `intl.formatMessage` instead of extracting it out?

File: src/elements/content-uploader/__tests__/ItemAction.test.js [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import noop from 'lodash/noop';
 import { shallow } from 'enzyme';
+import PlainButton from '../../../components/plain-button';
```
Comment (JChan106) [inline src/elements/content-uploader/__tests__/ItemAction.test.js:4]: Should we add test cases for null or undefined tooltip text?

File: src/elements/content-uploader/ItemAction.js (hunk 6 above)
Comment (QingyuChai) [reply src/elements/content-uploader/ItemAction.js:53]: I think Jason did that as a response to https://github.com/box/box-ui-elements/pull/3180#discussion_r1048812307

Comment (tjuanitas) [reply src/elements/content-uploader/ItemAction.js:53]: probably could go either way now. originally intended when there were multiple usages of `intl.formatMessage`

File: src/elements/content-uploader/__tests__/ItemAction.test.js (hunk 7 above)
Comment (jpan-box) [reply src/elements/content-uploader/__tests__/ItemAction.test.js:4]: yes, i wanted to but i think i should avoid it for now since:
* i don't know how we would end up with an undefined tooltip - only if there was no status?
* should keep this PR just to the a11y bits. We've got a big pile of components to go through and update so we want to do a pass with all just the a11y stuff first.

File: src/elements/content-uploader/ItemAction.js (hunk 5 above)
Comment (karelee7) [reply src/elements/content-uploader/ItemAction.js]: Ah I meant the `status === STATUS_STAGED`, but it sounds like you've got it! :)

**Review by JChan106 — APPROVED** [review]
